### PR TITLE
Fixing "problems in spack" as identified by repology

### PR DESCRIPTION
**Pull request by vsoch**
I've added spack to repology (went live in the middle of today) and as part of that, I stumbled on this entire three page of "spack problems"

https://repology.org/repository/spack/problems

And of course could not simply leave them there, so I went through each one and fixed the package. There were a few that should be further discussed in #25490 but for the most part these are small changes to update urls to use https, or update the perl package manager, or replace urls that are now 404. For a few sites that triggered errors (that didn't have https) the certificate setup might be strange but it's not broken, so I didn't change those. It looks like from the file count I was able to fix 379/397, which is pretty good!

Signed-off-by: vsoch <vsoch@users.noreply.github.com>

### CODE REVIEW

File: var/spack/repos/builtin/packages/connect-proxy/package.py
Comment (alalazo) [inline var/spack/repos/builtin/packages/connect-proxy/package.py]: What was the issue here?

File: var/spack/repos/builtin/packages/keepalived/package.py
Comment (alalazo) [inline var/spack/repos/builtin/packages/keepalived/package.py:15]: Did you check the `http` below? The `homepage` is information for users, but we actually use `url` to download tarballs - so if possible it'd be better to update that too.

**Review by alalazo — COMMENTED** [review]
Reviewed the first 63 packages. Just a couple of questions so far. I'll also ping @adamjstewart to have another pair of eyes on this one.

File: var/spack/repos/builtin/packages/connect-proxy/package.py
Comment (vsoch) [reply var/spack/repos/builtin/packages/connect-proxy/package.py]: This is actually a mistake (and I’ll fix after I wake up). I tried changing the package to a new repo and the build failed so I reverted back to the original recipe, and missed this hash detail. So safe to ignore, will fix.

File: var/spack/repos/builtin/packages/keepalived/package.py
Comment (vsoch) [reply var/spack/repos/builtin/packages/keepalived/package.py:15]: I did check quite a few of these (maybe not this one in particular) and found that for many http only worked. Since I did this all manually and it took hours to check and copy paste URLs I decided to scope my fixes to just homepages since that’s what repology cares about. I did have to manually open about 400 files and check and copy paste, yes, lol.

Comment (vsoch) [reply var/spack/repos/builtin/packages/keepalived/package.py:15]: I did check quite a few of these (maybe not this one in particular) and found that for many http only worked. Since I did this all manually and it took hours to check and copy paste URLs I decided to scope my fixes to just homepages since that’s what repology cares about. I did have to manually open about 400 files and check and copy paste, yes, lol.

Comment (alalazo) [reply var/spack/repos/builtin/packages/keepalived/package.py:15]: Ok, fair enough. This is anyhow an improvement over current state so LGTM.

File: var/spack/repos/builtin/packages/parmgridgen/package.py
Comment (alalazo) [inline var/spack/repos/builtin/packages/parmgridgen/package.py:18]: This one is weird. The homepage points to a Github repository that has v0.0.2 as the latest version dating 2015. The url refers to v1.0 so :man_shrugging:

**Review by alalazo — COMMENTED** [review]
Got to half of the perl packages, just another minor comment so far. 165/379

File: var/spack/repos/builtin/packages/boost/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/boost/package.py:22]: @hainest I seem to recall some issue that made us switch from https to http for boost, is that correct?

File: var/spack/repos/builtin/packages/grib-api/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/grib-api/package.py:12]: For these deprecated packages, should we add `deprecated=True` to each version?

@skosukhin

File: var/spack/repos/builtin/packages/gconf/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/gconf/package.py:12]: For these deprecated packages, should we add `deprecated=True` to each version?

File: var/spack/repos/builtin/packages/perl-html-parser/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/perl-html-parser/package.py:13]: I assume that if search.cpan.org is discouraged for homepages, it's also discouraged for URLs (even if repology doesn't test it). We should add fetching support like we did for `pypi` at some point. This can happen in a separate PR though.

File: var/spack/repos/builtin/packages/py-nose-cov/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-nose-cov/package.py]: You can remove this line, it's the default set by `pypi`

File: var/spack/repos/builtin/packages/py-pager/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pager/package.py]: You can remove this line, it's the default set by `pypi`

File: var/spack/repos/builtin/packages/py-pytest-pep8/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pytest-pep8/package.py]: You can remove this line, it's the default set by `pypi`

File: var/spack/repos/builtin/packages/py-wget/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-wget/package.py]: You can remove this line, it's the default set by `pypi`

File: var/spack/repos/builtin/packages/py-zc-buildout/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-zc-buildout/package.py]: You can remove this line, it's the default set by `pypi`

File: var/spack/repos/builtin/packages/r-amap/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/r-amap/package.py:17]: For R packages we really need a `cran` attribute similar to `pypi` that autopopulates all of these attributes. @glennpj are you interested in working on this?

File: var/spack/repos/builtin/packages/parmgridgen/package.py
Comment (vsoch) [reply var/spack/repos/builtin/packages/parmgridgen/package.py:18]: I wasn't sure what to do about this one - the original home page is 404 https://www-users.cse.umn.edu/~moulitsa/software.html. I chose the repo because it seems to be linked to those same docs: https://github.com/mrklein/ParMGridGen

**Review by adamjstewart — COMMENTED** [review]
I'm not particularly worried about homepage changes since Spack doesn't use them for anything. As long as you've checked any url changes to make sure that things can still be downloaded. If you ever want to go crazy on `url`s, Spack has a `spack url stats` command to see how many URLs still use http, and a `spack url list` command if you want to print them.

Personally I would leave the R and Perl changes out for now and do a separate PR that replaces everything with a `cran = '...'` or `cpan = '...'` similar to `pypi = '...'`. If @glennpj is busy I may be interested in working on this in the near future.

File: var/spack/repos/builtin/packages/connect-proxy/package.py
Comment (vsoch) [reply var/spack/repos/builtin/packages/connect-proxy/package.py]: okay this should be fixed.

File: var/spack/repos/builtin/packages/boost/package.py
Comment (vsoch) [reply var/spack/repos/builtin/packages/boost/package.py:22]: yeah looks like it redirects to the sourceforge page - I'll return back to http.

File: var/spack/repos/builtin/packages/grib-api/package.py
Comment (vsoch) [reply var/spack/repos/builtin/packages/grib-api/package.py:12]: Sure! Let me know which versions (or all)

File: var/spack/repos/builtin/packages/gconf/package.py
Comment (vsoch) [reply var/spack/repos/builtin/packages/gconf/package.py:12]: Sure thing, all versions?

File: var/spack/repos/builtin/packages/perl-html-parser/package.py
Comment (vsoch) [reply var/spack/repos/builtin/packages/perl-html-parser/package.py:13]: yeah that's a good idea!

File: var/spack/repos/builtin/packages/gconf/package.py
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/gconf/package.py:12]: Yep. If the package has a maintainer, just ping them first.

File: var/spack/repos/builtin/packages/boost/package.py
Comment (vsoch) [reply var/spack/repos/builtin/packages/boost/package.py:22]: This should be fixed.

File: var/spack/repos/builtin/packages/py-nose-cov/package.py
Comment (vsoch) [reply var/spack/repos/builtin/packages/py-nose-cov/package.py]: This should be fixed.

File: var/spack/repos/builtin/packages/py-pager/package.py
Comment (vsoch) [reply var/spack/repos/builtin/packages/py-pager/package.py]: This should be fixed.

File: var/spack/repos/builtin/packages/py-pytest-pep8/package.py
Comment (vsoch) [reply var/spack/repos/builtin/packages/py-pytest-pep8/package.py]: This should be fixed.

File: var/spack/repos/builtin/packages/py-wget/package.py
Comment (vsoch) [reply var/spack/repos/builtin/packages/py-wget/package.py]: This should be fixed.

File: var/spack/repos/builtin/packages/py-zc-buildout/package.py
Comment (vsoch) [reply var/spack/repos/builtin/packages/py-zc-buildout/package.py]: This should be fixed.

File: var/spack/repos/builtin/packages/boost/package.py
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/boost/package.py:22]: Not sure if this also applied to the list_url or not.

**Review by adamjstewart — DISMISSED** [review]
Python packages look good to me. If I get bored I'll add `cran` and `cpan` attributes for fetching someday.

File: var/spack/repos/builtin/packages/boost/package.py
Comment (vsoch) [reply var/spack/repos/builtin/packages/boost/package.py:22]: Maybe - I just changed it back.

**Comment by vsoch** [comment]
okay @adamjstewart I fixed that one boost issue, let me know if there is anything else you'd like before this is ready for merge!

**Comment by adamjstewart** [comment]
Did you add `deprecated=True` to the known deprecated packages?

**Comment by vsoch** [comment]
@adamjstewart I asked questions about those requests, they might be hidden now in the compressed comments. I asked specifically about versions I think.

**Comment by adamjstewart** [comment]
For at least one of those I said to just deprecate all versions if the entire package is deprecated, sorry for the confusion.

**Comment by vsoch** [comment]
No problem! All set.

File: var/spack/repos/builtin/packages/r-amap/package.py
Comment (glennpj) [reply var/spack/repos/builtin/packages/r-amap/package.py:17]: Sure, I actually planned on doing that when I do my next wave of CRAN package updates. I have not been pushing for that in PRs though.

Comment (glennpj) [reply var/spack/repos/builtin/packages/r-amap/package.py:17]: Just to make sure that I am clear about the goal, are we looking to have the `cran` attribute in all CRAN packages? The attribute already exists but its usage has not been standardized on as of yet.

Comment (vsoch) [reply var/spack/repos/builtin/packages/r-amap/package.py:17]: Maybe y'all want to open an issue on this instead of the thread getting lost in this PR? There are already so many comments I can't find ones that I'm looking for with that accordian thing GitHub is doing!

**Review by adamjstewart — APPROVED** [review]

File: var/spack/repos/builtin/packages/r-amap/package.py
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/r-amap/package.py:17]: Yes, the goal is to get a `cran` attribute for all packages. I didn't realize we already had support for it. I don't think an issue is necessary if you're already planning to do this in the future.

**Comment by vsoch** [comment]
YES!!! :partying_face:

**Comment by alalazo** [comment]
Given the commit message with which this landed in `develop`, I think @adamjstewart was hungry :laughing: For reference this is "[speck](https://en.wikipedia.org/wiki/Speck)"

**Comment by vsoch** [comment]
lol! So that was totally my mistake - I fixed the title from "speck" to "spack" but didn't update the commit message. Oups. 

And just looked at the link, ewsies! @adamjstewart you can have all the speck, do not save any for me :laughing:

**Pull request merged**
